tests/error/test-catalog.xml

```diff
@@ -159,6 +159,139 @@
+      cannot be serialized.</p></description>
+      <test-string>This sentence cannot be serialized.</test-string>
+      <result><assert-dynamic-error error-code="D05 D01"/></result>
+    </test-case> 
+  </test-set>
+
+  <test-set name="attribute-roots-plural">
+    <created by="MSM" on="2022-06-03"/>
+    <description><p>This grammar marks its start symbol as hidden;
+    when that symbol has multiple children marked as attributes, the
+    sentence cannot be serialized as well-formed
+    XML.</p></description>
+    <ixml-grammar>
+      -S: a, b, c, d.
+      @a: 'able'.
+      @b: 'baker'.
+      @c: 'charlie'.
+      d: 'dog'.
+    </ixml-grammar>
+    <!--* no grammar test, since some processors may detect
+	* the error statically. *-->
+    <test-case name="wb-not-wf">
+      <description><p>This input is grammatical but cannot be
+      serialized.</p></description>
+      <test-string>ablebakercharliedog</test-string>
+      <result><assert-dynamic-error error-code="D05 D01"/></result>
+    </test-case> 
+  </test-set>
+
+  <test-set name="well-balanced-xml">
+    <created by="MSM" on="2022-06-03"/>
+    <description><p>This grammar marks its start symbol as hidden;
+    when that symbol has multiple children, the sentence cannot be
+    serialized as well-formed XML.</p></description>
+    <ixml-grammar>
+      -S: a, b, c, d.
+      a: 'able'.
+      b: 'baker'.
+      c: 'charlie'.
+      d: 'dog'.
+    </ixml-grammar>
+    <!--* no grammar test, since some processors may detect
+	* the error statically. *-->
+    <test-case name="wb-not-wf">
+      <description><p>This input is grammatical but cannot be
+      serialized.</p></description>
+      <test-string>ablebakercharliedog</test-string>
+      <result><assert-dynamic-error error-code="D06 D01"/></result>
+    </test-case> 
+  </test-set>
+
+  <test-set name="rootless">
+    <created by="MSM" on="2022-06-03"/>
+    <description><p>This grammar marks all its nonterminal symbols as
+    hidden; since there is no outermost element, the parse tree cannot
+    be serialized as well-formed XML.</p></description>
+    <ixml-grammar>
+      -S: a, b, c, d.
+      -a: 'able'.
+      -b: 'baker'.
+      -c: 'charlie'.
+      -d: 'dog'.
+    </ixml-grammar>
+    <!--* no grammar test, since some processors may detect
+	* the error statically. *-->
+    <test-case name="textnode-at-root">
+      <description><p>This input is grammatical but cannot be
+      serialized.</p></description>
+      <test-string>ablebakercharliedog</test-string>
+      <result><assert-dynamic-error error-code="D06 D01"/></result>
+    </test-case> 
+  </test-set>
+
+  <test-set name="name-starting-xml">
+    <created by="MSM" on="2022-06-03"/>
+    <description><p>This grammar marks all its nonterminal symbols as
+    hidden; since there is no outermost element, the parse tree cannot
+    be serialized as well-formed XML.</p></description>
+    <ixml-grammar>
```

Comment:
I don't think this description accurately describes the test (cut-and-paste error, I think).